src/components/Home/HomeComponent.jsx

```diff
@@ -0,0 +1,125 @@
+import React, { Component } from 'react';
+import { Segment, Icon } from 'semantic-ui-react';
+import { withRouter} from 'react-router-dom';
+
+import HomeMap from './HomeMap';
+import HomeMapList from './HomeMapList';
+let popOpen = false;
+
+class Home extends Component {
+  state = { 
+    userLocation: { lat: '', lng: '' },
+    loading: false,
+    posts: [],
+    zoom: 12,
+    markerColor:'',
+    newBounds: null,
+    selectedMarkerId: '',
+    selectedPostId: '',
+    clickedMarkerId:'',
+    clickedPostId:'',
+  };
+  
+  getPosts = async () => {
+    if (this.state.newBounds !== null){
+      const response = await fetch('http://localhost:8081/api/posts?min_lat=' + this.state.newBounds.se.lat.toString() + '&max_lat=' + this.state.newBounds.ne.lat.toString() + '&min_lng=' + this.state.newBounds.sw.lng.toString() + '&max_lng=' + this.state.newBounds.ne.lng.toString());
+      const { posts } = await response.json();
+      this.setState({posts});
+    }
+  }
+    
+  componentDidMount = async () => {
+    await navigator.geolocation.watchPosition(
+      position => {
+        const { latitude, longitude } = position.coords;
+        
+        this.setState({
+          userLocation: {lat:latitude, lng:longitude},
+          loading: false
+        });
+      },
+      () => {
+        this.setState({ loading: false });
+      }
+    );
+  }
+  
+  changeBounds = async (props) => {
```

Comment:
No need to pass `props` as an argument when you can do `this.props`.

Reply:
you might need to take a look at this one

Reply:
Alright no worries. Let's take a look at this together once you visit.

Reply:
sounds good mang